src/vs/workbench/contrib/terminal/node/terminalProcess.ts

```diff
@@ -69,28 +69,32 @@ export class TerminalProcess implements ITerminalChildProcess, IDisposable {
 			experimentalUseConpty: useConpty
 		};
 
-		try {
-			this._ptyProcess = pty.spawn(shellLaunchConfig.executable!, shellLaunchConfig.args || [], options);
-			this._processStartupComplete = new Promise<void>(c => {
-				this.onProcessIdReady((pid) => {
-					c();
+		const filePath = path.basename(shellLaunchConfig.executable!);
+		fs.stat(filePath, (err, stats) => {
+			if (err === null) {
+				this._ptyProcess = pty.spawn(shellLaunchConfig.executable!, shellLaunchConfig.args || [], options);
+				this._processStartupComplete = new Promise<void>(c => {
+					this.onProcessIdReady((pid) => {
+						c();
+					});
 				});
-			});
-		} catch (error) {
-			// The only time this is expected to happen is when the file specified to launch with does not exist.
-			this._exitCode = 2;
-			this._queueProcessExit();
-			this._processStartupComplete = Promise.resolve(undefined);
-			return;
-		}
-		this._ptyProcess.on('data', (data) => {
+			}
+			else if (err.code === 'ENOENT') {
+				this._exitCode = SHELL_PATH_INVALID_EXIT_CODE;
+				this._queueProcessExit();
+				this._processStartupComplete = Promise.resolve(undefined);
+				return;
+			}
+		});
+
+		this._ptyProcess!.on('data', (data) => {
```

Comment:
I don't think this will work atm as _ptyProcess is delayed until after the stat but this runs immediately. Let move this and everything below in the ctor to inside the stat

Reply:
I think this is why the integration tests failed in CI

Reply:
Sorry that was my bad , should have checked the async behaviour , this is done now. Added a new function `setupPtyProcess` to make the code look cleaner :) ( Let me know if this is not needed )